kimchi/src/circuits/expr.rs

```diff
@@ -436,14 +436,14 @@ impl Variable {
             .as_ref()
             .ok_or(ExprError::LookupShouldNotBeUsed);
         match self.col {
-            Witness(i) => Ok(evals.w[i]),
-            Z => Ok(evals.z),
-            LookupSorted(i) => l.map(|l| l.sorted[i]),
-            LookupAggreg => l.map(|l| l.aggreg),
-            LookupTable => l.map(|l| l.table),
-            LookupRuntimeTable => l.and_then(|l| l.runtime.ok_or(ExprError::MissingRuntime)),
-            Index(GateType::Poseidon) => Ok(evals.poseidon_selector),
-            Index(GateType::Generic) => Ok(evals.generic_selector),
+            Witness(i) => Ok(evals.w[i][0]),
+            Z => Ok(evals.z[0]),
+            LookupSorted(i) => l.map(|l| l.sorted[i][0]),
+            LookupAggreg => l.map(|l| l.aggreg[0]),
+            LookupTable => l.map(|l| l.table[0]),
+            LookupRuntimeTable => l.and_then(|l| l.get_runtime().ok_or(ExprError::MissingRuntime)),
+            Index(GateType::Poseidon) => Ok(evals.poseidon_selector[0]),
+            Index(GateType::Generic) => Ok(evals.generic_selector[0]),
```

Comment:
this feels very unsafe to me. If we're going to this route I think it'd be better to have two distinct type, one with a vec and one without

Reply:
Perhaps use a `is_chunked()` function to check the situation and handle any errors?  Maybe I missed somewhere were you already do it.